lms/djangoapps/edraak_university/views.py

```diff
@@ -130,13 +183,31 @@ class UniversityIDUpdateView(CourseContextMixin, generic.UpdateView):
     template_name = 'edraak_university/instructor/update.html'
 
     # The email and full_name fields are written directly in the `update.html` file.
-    fields = ('university_id', 'section_number',)
+    fields = ('university_id',)
 
     def get_success_url(self):
         return reverse('edraak_university_id_list', kwargs={
             'course_id': self.get_course_key(),
         })
 
+    def form_valid(self, form):
+        instance = form.save(commit=False)
+        instance.save()
+        cohort_id = self.request.POST['cohort']
+        cohort = CourseUserGroup.objects.get(id=cohort_id)
+        try:
+            add_user_to_cohort(cohort, instance.user.email)
+        except ValueError:
```

Comment:
decorator on dispatch method `@method_decorator(ensure_valid_course_key)` should be doing the same thing